### PR TITLE
[@types/web3] :arrow_up: Solidity compiler defines stateMutability for ABIDefinition

### PR DESCRIPTION
**Pull request by thibmeu**
Reference regarding ABIDefinition has been updated. It is available here https://solidity.readthedocs.io/en/v0.5.0/abi-spec.html#json

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://solidity.readthedocs.io/en/v0.5.0/abi-spec.html#json
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by levino — CHANGES_REQUESTED** [review]
Please add a couple of regression tests.

File: types/web3/web3-tests.ts [hunk 1]
```diff
@@ -125,6 +139,95 @@ web3.eth.personal.unlockAccount(
...
+        constant: false,
+        payable: true,
+        stateMutability: "payable",
+        inputs: [],
+        name: "removeOwner",
+        outputs: [],
+        type: "function"
+    },
+];
+const myContractNewAbi = new web3.eth.Contract(myContractNewAbiStandard);
```
Comment (Levino) [inline types/web3/web3-tests.ts]: Thank you. Is it possible to create a second test file for this?

File: types/web3/web3-tests.ts [hunk 2]
```diff
@@ -20,34 +21,47 @@ const storage: Promise<string> = web3.eth.getStorageAt(contractAddress, 0);
...
+    .on("receipt", (txReceipt: TransactionReceipt) => {
+    });
+const sendTransactionTxReceipt3: PromiEvent<TransactionReceipt> = web3.eth.sendTransaction({to: "0x1"})
+    .on("confirmation", (confNumber: number, receipt: TransactionReceipt) => {
+    });
+const sendTransactionTxReceipt4: PromiEvent<TransactionReceipt> = web3.eth.sendTransaction({to: "0x1"})
+    .on("receipt", (txReceipt: TransactionReceipt) => {
+    })
+    .on("confirmation", (confNumber: number, receipt: TransactionReceipt) => {
+    });
```
Comment (Levino) [inline types/web3/web3-tests.ts]: Please undo all these formatting changes. What is this? Prettier?

**Review by levino — CHANGES_REQUESTED** [review]

File: types/web3/web3-tests.ts (hunk 1 above)
Comment (thibmeu) [reply types/web3/web3-tests.ts]: The only file tested (linted) by `npm run test web3` is `web3-tests.ts`. Therefore I have added `test/abi-tests.ts`, containing an example of old and new abi standard.

**Comment by Levino** [comment]
Very good work, thank you. The only thing that remains is that I would suggest squashing all this into one commit before merging. I do not think that when I approve the PR it will get squashed and merged. Do you want to squash @thibmeu ? If not then I will just approve and we leave the intermediate commits in, which is not perfect but doable.

**Comment by thibmeu** [comment]
I have just squashed them. @Levino Thank you for your feedback, PR looks much better.

**Review by levino — APPROVED** [review]

**Review by phra — APPROVED** [review]

**Pull request merged**